saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/FavoritesDelegate.kt

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.savedsites.impl
+
+import com.duckduckgo.app.global.DispatcherProvider
+import com.duckduckgo.app.global.formatters.time.*
+import com.duckduckgo.di.scopes.*
+import com.duckduckgo.savedsites.api.models.*
+import com.duckduckgo.savedsites.impl.sync.*
+import com.duckduckgo.savedsites.store.*
+import com.duckduckgo.sync.api.*
+import com.squareup.anvil.annotations.*
+import io.reactivex.*
+import java.util.*
+import javax.inject.Inject
+import kotlinx.coroutines.flow.*
+import timber.log.*
+
+interface FavoritesDelegate {
```

Comment:
already mentioned on a previous PR. This is a delegate, public api is already documented. I don't think we need to add any docs to this one since it's internal.